4.analyze_data/notebooks/UMAP/nbconverted/2.find_sc.py

```diff
@@ -0,0 +1,217 @@
+            "Metadata_Well",
+            "Metadata_Site",
+            "Metadata_Cells_Location_Center_X",
+            "Metadata_Cells_Location_Center_Y",
+            "UMAP0",
+            "UMAP1",
+        ]
+    )
+
+
+# ## Read in `yaml` file with settings for each single cell type
+
+# In[3]:
+
+
+# load in plate information
+dictionary_path = pathlib.Path("./image_settings.yaml")
+with open(dictionary_path) as file:
+    cell_info_dictionary = yaml.load(file, Loader=yaml.FullLoader)
+
+# view the dictionary to assess that all info is added correctly
+pprint.pprint(cell_info_dictionary, indent=4)
+
+
+# ## Find single cells from each cell type and treatment from UMAP (plate 3)
+
+# In[4]:
+
+
+for crop_cell, info in cell_info_dictionary.items():
+    # Define your filter conditions
+    condition = (
+        (UMAP_plate3_df['Metadata_cell_type'] == crop_cell.split('-')[0]) &
+        (UMAP_plate3_df['Metadata_treatment'] == crop_cell.split('-')[1]) &
+        (UMAP_plate3_df['UMAP0'].between(info["UMAP0"][0], info["UMAP0"][1]) &
+        (UMAP_plate3_df['UMAP1'].between(info["UMAP1"][0], info["UMAP1"][1]))
+    ))
+
+    # Apply the filter
+    filtered_df = UMAP_plate3_df[condition]
+
+    # Apply the filter
+    filtered_df = UMAP_plate3_df[condition]
+
+    # Select and display only the specific columns
+    selected_columns = [
+        'Metadata_cell_type',
+        'Metadata_treatment',
+        'Metadata_Plate',
+        'Metadata_Well',
+        'Metadata_Site',
+        'Metadata_Cells_Location_Center_X',
+        'Metadata_Cells_Location_Center_Y',
+        'UMAP0',
+        'UMAP1'
+    ]
+
+    filtered_df = filtered_df[selected_columns]
+
+    # Randomly select a row
+    random_row = filtered_df.sample(n=1, random_state=info["Random state"])  
+
+    # Add the selected row to the DataFrame
+    df = df.append(random_row, ignore_index=True)
+
+    # Assign the direction to the UMAP_direction column for the last added row
+    df.at[df.index[-1], "UMAP_direction"] = crop_cell.split('-')[2]
+
+    # Save the updated DataFrame to the CSV file
+    df.to_csv(csv_file_path, index=False)
+
+    # Create a filename based on Metadata_Plate, Metadata_Well, Metadata_Site
+    plate = random_row['Metadata_Plate'].values[0]
+    well = random_row['Metadata_Well'].values[0]
+    site = random_row['Metadata_Site'].values[0]
+
+    # Initialize a list to store file paths
+    file_paths = []
+
+    # Create 5 different file paths with "d0" through "d4" suffixes
+    for i in range(5):
+        filename = f"{images_dir}/{plate}_{well}{site}d{i}_illumcorrect.tiff"
+        file_paths.append(filename)
+
+    print("Randomly selected row:")
+    print(random_row)
+    print("Generated filenames:")
+    for path in file_paths:
+        print(path)
+
+    # Initialize empty lists to store the images for each channel
+    blue_channel = []
+    green_channel = []
+    red_channel = []
+
+    # Iterate over files in the directory
```

Comment:
Are you separating out the different channel images here? Maybe add more documentation inline for this?

Reply:
Ah this comment isn't accurate, thanks for pointing it out! I will revise it.